contracts/account/modules/ERC7579SocialRecovery.sol

```diff
@@ -0,0 +1,371 @@
+        return (exist_, property_.toUint48());
+    }
+
+    function getAccountConfigs() external view returns (RecoveryConfigArg memory recoveryConfigArg) {
+        return getAccountConfigs(msg.sender);
+    }
+
+    function getAccountConfigs(
+        address account
+    ) public view virtual returns (RecoveryConfigArg memory recoveryConfigArg) {
+        AccountConfig storage config = _configs[account];
+
+        bytes[] memory guardians = new bytes[](config.guardians.length());
+        for (uint256 i = 0; i < guardians.length; ++i) {
+            (bytes memory identity, uint256 property) = config.guardians.at(i);
+            guardians[i] = _formatGuardian(property.toUint64(), identity);
+        }
+
+        ThresholdConfig[] memory thresholds = new ThresholdConfig[](config.thresholds.length());
+        for (uint256 i = 0; i < thresholds.length; ++i) {
+            Checkpoints.Checkpoint160 memory ckpt = config.thresholds.at(i.toUint32());
+            thresholds[i].threshold = ckpt._key.toUint64();
+            thresholds[i].lockPeriod = ckpt._value.toUint48();
+        }
+
+        return RecoveryConfigArg({verifier: config.verifier, guardians: guardians, thresholds: thresholds});
+    }
+
+    function getRecoveryNonce() external view returns (uint256 nonce) {
+        return nonces(msg.sender);
+    }
+
+    function getRecoveryStatus() external view returns (bool isRecovering, uint48 expiryTime) {
+        return getRecoveryStatus(msg.sender);
+    }
+
+    function getRecoveryStatus(address account) public view virtual returns (bool isRecovering, uint48 expiryTime) {
+        expiryTime = _configs[account].expiryTime;
+        isRecovering = expiryTime != 0;
+    }
+
+    /****************************************************************************************************************
+     *                                          Social recovery - internal                                          *
+     ****************************************************************************************************************/
+    function _overrideConfig(address account, RecoveryConfigArg calldata recoveryConfigArgs) internal virtual {
+        _clearConfig(account);
+
+        AccountConfig storage config = _configs[account];
+
+        // verifier
+        config.verifier = recoveryConfigArgs.verifier == address(0) ? address(this) : recoveryConfigArgs.verifier;
+
+        // guardians
+        for (uint256 i = 0; i < recoveryConfigArgs.guardians.length; ++i) {
+            (uint64 property, bytes calldata identity) = _parseGuardian(recoveryConfigArgs.guardians[i]);
+            config.guardians.set(identity, property);
+        }
+
+        // threshold
+        for (uint256 i = 0; i < recoveryConfigArgs.thresholds.length; ++i) {
+            config.thresholds.push(
+                recoveryConfigArgs.thresholds[i].threshold,
+                recoveryConfigArgs.thresholds[i].lockPeriod
+            );
+        }
```

Comment:
Should require sorting by threshold?

Reply:
This is not an array push, its a checkpoint.trace push. It already includes a sorting check internally

- [push](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/Checkpoints.sol#L39-L45)
  - [_insert](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/Checkpoints.sol#L39-L45)
    - if (length > 0)
      - [if (lastKey > key) revert](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/Checkpoints.sol#L147-L149)